terraform/spec/dependabot/terraform/file_parser_spec.rb

```diff
@@ -9,336 +9,275 @@
 RSpec.describe Dependabot::Terraform::FileParser do
   it_behaves_like "a dependency file parser"
 
-  let(:files) { [terraform_file] }
-  let(:terraform_file) do
-    Dependabot::DependencyFile.new(name: "main.tf", content: terraform_body)
-  end
-  let(:terraform_body) do
-    fixture("config_files", terraform_fixture_name)
-  end
-  let(:terraform_fixture_name) { "git_tags.tf" }
-  let(:parser) { described_class.new(dependency_files: files, source: source) }
-  let(:source) do
-    Dependabot::Source.new(
-      provider: "github",
-      repo: "gocardless/bump",
-      directory: "/"
-    )
-  end
-
-  describe "parse" do
-    subject(:dependencies) { parser.parse }
-
-    context "with registry sources" do
-      let(:terraform_fixture_name) { "registry.tf" }
-
-      its(:length) { is_expected.to eq(5) }
-
-      context "that are invalid" do
-        let(:terraform_fixture_name) { "invalid_registry.tf" }
-
-        it "raises a helpful error" do
-          expect { parser.parse }.
-            to raise_error(Dependabot::DependencyFileNotEvaluatable) do |err|
-              expect(err.message).
-                to eq("Invalid registry source specified: 'consul/aws'")
-            end
-        end
+  subject(:parser) { described_class.new(dependency_files: files, source: source) }
+
+  let(:files) { [] }
+  let(:source) { Dependabot::Source.new(provider: "github", repo: "gocardless/bump", directory: "/") }
+
+  describe "#parse" do
+    subject { parser.parse }
+
+    context "with an invalid registry source" do
+      let(:files) do
+        [
+          Dependabot::DependencyFile.new(
+            name: "main.tf",
+            content: fixture("projects", "invalid_registry", "main.tf")
```

Comment:
Noice: project fixtures!

Idea: maybe `project_name` could be a scoped variable?
I think the contents of `files` could be derived from something like `Dir.glob("fixtures/#{project_name}/*").map{ |p| DependencyFile.new(...) }`. That's consistent with what the [FileFetcher would do](https://github.com/dependabot/dependabot-core/blob/5e3d19ed811c76f70dda0b2ff3e6aeeaf856fe9e/terraform/lib/dependabot/terraform/file_fetcher.rb#L35) - return all the `.tf` files.

Reply:
We found a helper called `project_dependency_files` that seems to accomplish this same goal. Thank you for calling out this suggestion.